<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>robotcontrolClass</class>
 <widget class="QMainWindow" name="robotcontrolClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1002</width>
    <height>649</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_12">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_6">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_12">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <widget class="QGroupBox" name="groupBox">
            <property name="title">
             <string>TCP Offset</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_9">
             <item>
              <widget class="QTableWidget" name="offsettable">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="editTriggers">
                <set>QAbstractItemView::NoEditTriggers</set>
               </property>
               <row>
                <property name="text">
                 <string>X</string>
                </property>
               </row>
               <row>
                <property name="text">
                 <string>Y</string>
                </property>
               </row>
               <row>
                <property name="text">
                 <string>Z</string>
                </property>
               </row>
               <row>
                <property name="text">
                 <string>T</string>
                </property>
               </row>
               <column>
                <property name="text">
                 <string>Output</string>
                </property>
               </column>
               <column>
                <property name="text">
                 <string>Input</string>
                </property>
               </column>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <widget class="QPushButton" name="setoffset_btn">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>Set Offset</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="loadfile">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>Load File</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_9">
            <item>
             <widget class="QComboBox" name="axiscombox">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <item>
               <property name="text">
                <string>Axis1</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Axis2</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Axis3</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Axis4</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Axis5</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Axis6</string>
               </property>
              </item>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_8">
              <item>
               <widget class="QPushButton" name="jog_p_btn">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string>+</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="jog_n_btn">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string>-</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>Joint Position</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_10">
           <item>
            <widget class="QTableWidget" name="jointtable">
             <property name="editTriggers">
              <set>QAbstractItemView::NoEditTriggers</set>
             </property>
             <row>
              <property name="text">
               <string>Axis1</string>
              </property>
             </row>
             <row>
              <property name="text">
               <string>Axis2</string>
              </property>
             </row>
             <row>
              <property name="text">
               <string>Axis3</string>
              </property>
             </row>
             <row>
              <property name="text">
               <string>Axis4</string>
              </property>
             </row>
             <row>
              <property name="text">
               <string>Axis5</string>
              </property>
             </row>
             <row>
              <property name="text">
               <string>Axis6</string>
              </property>
             </row>
             <column>
              <property name="text">
               <string>Output</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Input</string>
              </property>
             </column>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_17">
             <item>
              <widget class="QLabel" name="label_25">
               <property name="text">
                <string>V:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSlider" name="horizontalSlider_3">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="doubleSpinBox_33">
               <property name="suffix">
                <string>%</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="gridGroupBox">
          <property name="title">
           <string>TCP and TOV</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_11">
           <item>
            <widget class="QTableWidget" name="tcptable">
             <property name="editTriggers">
              <set>QAbstractItemView::NoEditTriggers</set>
             </property>
             <row>
              <property name="text">
               <string>X</string>
              </property>
             </row>
             <row>
              <property name="text">
               <string>Y</string>
              </property>
             </row>
             <row>
              <property name="text">
               <string>Z</string>
              </property>
             </row>
             <row>
              <property name="text">
               <string>A</string>
              </property>
             </row>
             <row>
              <property name="text">
               <string>B</string>
              </property>
             </row>
             <row>
              <property name="text">
               <string>C</string>
              </property>
             </row>
             <column>
              <property name="text">
               <string>Output</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Input</string>
              </property>
             </column>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_16">
             <item>
              <widget class="QRadioButton" name="radioButton">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>Input</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QRadioButton" name="radioButton_2">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>Forward</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QRadioButton" name="radioButton_3">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>Down</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_10">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QPushButton" name="start_btn">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Start</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="stop_btn">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Stop</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="servo_on_btn">
            <property name="text">
             <string>Servo ON</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="servo_off_btn">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Servo OFF</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="exit_btn">
            <property name="text">
             <string>Exit</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QLineEdit" name="ed_editor"/>
          </item>
          <item>
           <widget class="QListWidget" name="list_nc"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <widget class="QGroupBox" name="groupBox_4">
            <property name="title">
             <string>Gripper Control</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_2">
               <item>
                <widget class="QPushButton" name="pushButton_3">
                 <property name="enabled">
                  <bool>false</bool>
                 </property>
                 <property name="text">
                  <string>Open</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="pushButton_4">
                 <property name="enabled">
                  <bool>false</bool>
                 </property>
                 <property name="text">
                  <string>Close</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_3">
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout">
                 <item>
                  <widget class="QLabel" name="label_21">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="text">
                    <string>P:</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QDoubleSpinBox" name="doubleSpinBox_30">
                   <property name="suffix">
                    <string>mm</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
               <item>
                <widget class="QPushButton" name="pushButton_5">
                 <property name="enabled">
                  <bool>false</bool>
                 </property>
                 <property name="text">
                  <string>Move</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_4">
               <item>
                <widget class="QLabel" name="label_22">
                 <property name="text">
                  <string>V:</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QSlider" name="horizontalSlider">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QDoubleSpinBox" name="doubleSpinBox_29">
                 <property name="suffix">
                  <string>%</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_5">
               <item>
                <widget class="QLabel" name="label_23">
                 <property name="text">
                  <string>F:</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QSlider" name="horizontalSlider_2">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QDoubleSpinBox" name="doubleSpinBox_31">
                 <property name="suffix">
                  <string>%</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_6">
               <item>
                <widget class="QLabel" name="label_24">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>Delay</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QDoubleSpinBox" name="doubleSpinBox_32">
                 <property name="suffix">
                  <string>sec</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_8">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_13">
              <item>
               <widget class="QPushButton" name="move_btn">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string>Move</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="joint_ctr_btn">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string>Joint Control</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_14">
              <item>
               <widget class="QPushButton" name="p2p_ctr_btn">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string>P2P Control</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="cp_ctr_btn">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string>CP Control</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_15">
              <item>
               <widget class="QPushButton" name="enter_btn">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string>Enter</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="mode_btn">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string>Change Mode</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_11">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_18">
              <item>
               <widget class="QPushButton" name="readtxt_btn">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string>Read TXT</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="paint_btn">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string>Paint Taiwan</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="home_btn">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string>Home</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
